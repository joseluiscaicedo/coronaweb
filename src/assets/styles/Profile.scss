
.ProfileContainer__img{
  border-radius: 50%;
  width: 3em;

}