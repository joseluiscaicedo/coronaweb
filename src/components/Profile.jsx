import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import '../assets/styles/Profile.scss';

const Profile = () => {
  const { user, isAuthenticated, isLoading } = useAuth0();

  if (isLoading) {
    return <div>Loading...</div>;
  }

  return (
    isAuthenticated && (
      <div className='ProfileContainer'>
        <img
          className='ProfileContainer__img'
          src={user.picture}
          alt={user.name}
        />
      </div>
    )
  );
};

export default Profile;
