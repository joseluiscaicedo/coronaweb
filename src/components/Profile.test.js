import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Profile from './Profile'

jest.mock('@auth0/auth0-react', () => ({
  useAuth0: ()=>{return {
    'isAuthenticated':true,
    'isLoading':true,
    'user':{}
  }},
}))

describe('validate Profile', () =>{

  test('Generates Snapshot for Profile Component', () => {
    const { container } = render(
      <Profile />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test('Validate by Text', () => {
    const component =render( <Profile /> );
    const SpanElement=component.getByText('Loading...')
   expect(SpanElement).toBeInTheDocument();
 });
})