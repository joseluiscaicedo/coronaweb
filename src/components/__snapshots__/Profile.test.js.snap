// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate Profile Generates Snapshot for Profile Component 1`] = `
<div>
  Loading...
</div>
`;
